agent/ec2/ec2_metadata_client.go

```diff
@@ -124,17 +137,36 @@ func (c *ec2MetadataClientImpl) ResourceServiceUrl(path string) string {
 func (c *ec2MetadataClientImpl) ReadResource(path string) ([]byte, error) {
 	endpoint := c.ResourceServiceUrl(path)
 
-	resp, err := c.client.Get(endpoint)
+	var err error
+	var resp *http.Response
```

Comment:
I don't like this model of sharing variables among functions (makes resource handling/owning/closing) ambiguous. But, I guess you are doing it this way because of the arguments expected by RetryNWithBackoff.